subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -325,8 +325,17 @@ static bool schedule_send(struct bt_mesh_ext_adv *adv)
 
 	timestamp = adv->timestamp;
 
-	if (atomic_test_and_clear_bit(adv->flags, ADV_FLAG_PROXY)) {
+	/* Because the `ADV_FLAG_PROXY` flag has been set in advance, so should
+	 * check `BT_ADV_ENABLED` flag indicate the advertising actually enabled.
+	 *
+	 * After these conditions are met, we can close the Proxy Advertising,
+	 * otherwise we can only postpone until all hci cmd of the Proxy are completed.
+	 */
+	if (atomic_test_bit(adv->flags, ADV_FLAG_PROXY) &&
+	    atomic_test_bit(adv->instance->flags, BT_ADV_ENABLED)) {
```

Comment:
Is there a way to not access internal fields of `bt_le_ext_adv`? It would be better if we don't add this dependency to mesh code.

Reply:
yeah，I'm not found same function todo this.

Reply:
An ugly suggestion would be to add `ADV_FLAG_PROXY_SCHEDULED`, set it before `bt_mesh_adv_gatt_send()`, and then check in `connected` callback (I guess this is the only place where it is actually needed).